File: 198-house-robber/house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n= (int)nums.size();
        vector<int> dp(n+1);
        dp[1]=nums[0];
        for(int i=2;i<=n;++i){
            dp[i]=max(dp[i-1],dp[i-2]+nums[i-1]);
        }
        return dp[n];
    }
};

File: 198-house-robber/house-robber.java
class Solution {
    public int rob(int[] arr) {
        int n= arr.length;
        if(n==1) return arr[0];
        else if(n==2) return Math.max(arr[0],arr[1]);
        int[] dp= new int[n];
        dp[0]= arr[0]; dp[1]= Math.max(arr[0],arr[1]);
        for(int i=2;i<n;i++){
            dp[i]= Math.max(dp[i-1],arr[i]+dp[i-2]);
        }
        return dp[n-1];
    }
}